src/components/Home/Home.tsx

```diff
@@ -8,16 +8,17 @@ import type { DeviceType } from "@/types";
 
 export type Props = {
   devices: DeviceType[];
+  onNavigateToDevice: (deviceKey: DeviceType["key"]) => void;
 };
 
-const Home: FC<Props> = ({ devices }) => {
+const Home: FC<Props> = ({ devices, onNavigateToDevice }) => {
   return (
     <Container>
       {isEmpty(devices) ? (
         <NotFound text="No devices found" />
       ) : map(devices, (device) => (
         <Fragment key={device.key}>
-          <DeviceItem device={device}  />
+          <DeviceItem device={device} onClickTitle={() => onNavigateToDevice(device.key)} />
```

Comment:
Ideally callbacks should use `useCallback` to reduce the number of re-renders should they occur.

Reply:
Could you show me how better to use `useCallback` in this case please

Reply:
source: https://react.dev/reference/react/useCallback

It's like `useMemo`, e.g.
```
...
const handleTitleClick= useCallback(() => {
    onNavigateToDevice(device.key)
  }, [device.key]);
...
<DeviceItem device={device} onClickTitle={handleTitleClick} />
... 
```